File: backend/app/routes/resources.py
from fastapi import APIRouter
from app.data.resources import get_learning_resources

router = APIRouter()


@router.get("")
def get_resources(subject: str, chapter: str, grade: str = "Grade 9"):
    """Return curated learning resources for the selected subject/chapter."""
    return {
        "success": True,
        "resources": get_learning_resources(subject, chapter, grade)
    }

File: frontend/src/api/resources.js
const API_BASE_URL =
  import.meta.env.VITE_API_BASE_URL ||
  "http://localhost:8000";

export async function getResources(subject, chapter, grade = "Grade 9") {
  /** Load curated free learning resources for one subject/chapter pair. */
  const params = new URLSearchParams({
    grade,
    subject,
    chapter,
  });

  const response = await fetch(`${API_BASE_URL}/api/resources?${params.toString()}`);

  if (!response.ok) {
    throw new Error("Failed to load resources");
  }

  return response.json();
}
